usaspending_api/etl/es_etl_helpers.py

```diff
@@ -0,0 +1,465 @@
+    '''
+    Populates the formatted strings defined globally in this file to create the desired SQL
+    '''
+    update_date_str = UPDATE_DATE_SQL.format(config['starting_date'].strftime('%Y-%m-%d'))
+    award_type_str = ''
+    if config['award_category']:
+        if config['award_category'] == 'contract':
+            award_type_str = CONTRACTS_IDV_SQL.format(config['award_category'])
+        else:
+            award_type_str = CATEGORY_SQL.format(config['award_category'])
+
+    copy_sql = COPY_SQL.format(
+        fy=config['fiscal_year'],
+        update_date=update_date_str,
+        award_category=award_type_str,
+        filename=filename)
+
+    count_sql = COUNT_SQL.format(
+        fy=config['fiscal_year'],
+        update_date=update_date_str,
+        award_category=award_type_str,
+    )
+
+    if deleted_ids and config['provide_deleted']:
+        id_list = ','.join(["('{}')".format(x) for x in deleted_ids.keys()])
+        id_sql = CHECK_IDS_SQL.format(id_list=id_list, fy=config['fiscal_year'])
+    else:
+        id_sql = None
+
+    return copy_sql, id_sql, count_sql
+
+
+def execute_sql_statement(cmd, results=False, verbose=False):
+    ''' Simple function to execute SQL using the Django DB connection'''
+    rows = None
+    if verbose:
+        print(cmd)
+    with connection.cursor() as cursor:
+        cursor.execute(cmd)
+        if results:
+            rows = db_rows_to_dict(cursor)
+    return rows
+
+
+def db_rows_to_dict(cursor):
+    ''' Return a dictionary of all row results from a database connection cursor '''
+    columns = [col[0] for col in cursor.description]
+    return [
+        dict(zip(columns, row))
+        for row in cursor.fetchall()
+    ]
+
+
+def download_db_records(fetch_jobs, done_jobs, config):
+    while not fetch_jobs.empty():
+        if done_jobs.full():
+            printf({'msg': 'Waiting 60s reduce temporary disk space used', 'f': 'Download'})
+            sleep(60)
+        else:
+            start = perf_counter()
+            job = fetch_jobs.get_nowait()
+            printf({'msg': 'Preparing to download "{}"'.format(job.csv), 'job': job.name, 'f': 'Download'})
+
+            sql_config = {
+                'starting_date': config['starting_date'],
+                'fiscal_year': job.fy,
+                'award_category': job.category,
+                'provide_deleted': config['provide_deleted']
+            }
+            copy_sql, _, count_sql = configure_sql_strings(sql_config, job.csv, [])
+
+            if os.path.isfile(job.csv):
+                os.remove(job.csv)
+
+            job.count = download_csv(count_sql, copy_sql, job.csv, job.name, config['verbose'])
+            done_jobs.put(job)
+            printf({
+                'msg': 'CSV "{}" copy took {} seconds'.format(job.csv, perf_counter() - start),
+                'job': job.name,
+                'f': 'Download'
+            })
+            sleep(1)
+
+    done_jobs.put(DataJob(None, None, None, None, None))
```

Comment:
Can you insert a comment here to explain this?

Reply:
Will do